lib/internal/crypto/webidl.js

```diff
@@ -0,0 +1,740 @@
+'use strict';
+
+// Adapted from the following sources
+// - https://github.com/jsdom/webidl-conversions
+//   Copyright Domenic Denicola. Licensed under BSD-2-Clause License.
+//   Original license at https://github.com/jsdom/webidl-conversions/blob/master/LICENSE.md.
+// - https://github.com/denoland/deno
+//   Copyright Deno authors. Licensed under MIT License.
+//   Original license at https://github.com/denoland/deno/blob/main/LICENSE.md.
+// Changes include using primordials and stripping the code down to only what
+// WebCryptoAPI needs.
+
+const {
+  ArrayBufferIsView,
+  ArrayBufferPrototype,
+  ArrayPrototypePush,
+  ArrayPrototypeSort,
+  globalThis,
+  MathPow,
+  MathTrunc,
+  Number,
+  NumberIsFinite,
+  ObjectAssign,
+  ObjectDefineProperty,
+  ObjectPrototypeIsPrototypeOf,
+  ReflectHas,
+  SafeArrayIterator,
+  SafeSet,
+  String,
+  SymbolIterator,
+  TypedArrayPrototypeGetSymbolToStringTag,
+  TypeError,
+} = primordials;
+
+const { kEmptyObject } = require('internal/util');
+
+const { CryptoKey } = require('internal/crypto/webcrypto');
+
+const { SharedArrayBuffer } = globalThis;
```

Comment:
I'm surprised the linter didn't catch that.

```suggestion
  TypeError,
  globalThis: {
    SharedArrayBuffer,
  },
} = primordials;

const { kEmptyObject } = require('internal/util');

const { CryptoKey } = require('internal/crypto/webcrypto');
```

Reply:
The linter actually told me to do it this way when i just used the global as a global.

Reply:
Ah, because globalThis is already coming from primordials. I think the current code is fine.